packages/eas-cli/src/metadata/config.ts

```diff
@@ -12,6 +18,83 @@ export interface MetadataConfig {
   apple?: AppleMetadata;
 }
 
+/**
+ * Resolve the dynamic config from the user.
+ * It supports methods, async methods, or objects (json).
+ */
+async function resolveDynamicConfigAsync(configFile: string): Promise<unknown> {
+  const userConfigContent = await import(configFile);
+  let userConfig = userConfigContent.default || userConfigContent;
+
+  if (typeof userConfig === 'function') {
+    userConfig = await userConfig();
+  }
+
+  if (typeof userConfig === 'object' && userConfig !== null && !Array.isArray(userConfig)) {
+    return userConfig;
+  }
+
+  throw new MetadataValidationError(
+    `Metadata store config returned an unkown value, should be an object: "${userConfig}"`
+  );
+}
+
+/**
+ * Get the static configuration file path, based on the metadata context.
+ * This uses any custom name provided, but swaps out the extension for `.json`.
+ */
+export function getStaticConfigFile({
+  projectDir,
+  metadataPath,
+}: Pick<MetadataContext, 'projectDir' | 'metadataPath'>): string {
```

Comment:
nit
If you are passing the entire context either way, then I would prefer:
```suggestion
}: MetadataContext): string {
```
or if you want to narrow down the type (and pass only those 2 props on call-site)
```suggestion
}: { projectDir: string, metadataPath: string }): string {
```

Reply:
I think I used `MetadataContext` when writing this first. But, when I worked on the tests, it was kind of annoying to mock out the whole `MetadataContext` every time. So I think this is how I got to that point: to make testing easier.

Let's go for the `{ projectDir, metadataPath }` option.